tests/ui/mem_replace.rs

```diff
@@ -8,14 +8,36 @@
 // except according to those terms.
 
 // run-rustfix
-#![allow(unused_imports)]
+#![allow(unused_imports, deprecated, invalid_value)]
 #![warn(clippy::all, clippy::style, clippy::mem_replace_option_with_none)]
 
 use std::mem;
 
+fn might_panic(v: Vec<i32>) -> Vec<i32> { v }
+
 fn main() {
     let mut an_option = Some(1);
     let _ = mem::replace(&mut an_option, None);
     let an_option = &mut Some(1);
     let _ = mem::replace(an_option, None);
+
+    let mut v = vec![0i32; 4];
+    // the following is UB if `might_panic` panics
+    unsafe {
+        let taken_v = mem::replace(&mut v, mem::uninitialized());
```

Comment:
This statement (and the one below) should emit an error, but the `*.stderr` file is unchanged. 

Since this lint is not auto applicable it also might be necessary to extract the test into a new file. (`run-rustfix` will probably error)